### DISCUSSION TITLE
Tint Nolia desk lamp - 404117 - No converter available for 'State'

### DISCUSSION THREAD

**Discussion by lastfreeusernameicouldfind** [Koenkk/zigbee2mqtt#20077] [Devices]
Hi all,

I got a new tint Nolia desk lamp (404117) and tried to onboard it to zigbee2mqtt. My goal would be to be able to turn the device on and off via mqtt messages. 
I carefully followed https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html but somehow I do not get it to work. Whats working is that if I am turning it on/off on the device with the built in hardware toggle the status message is received by zigbee2mqtt.
```
Received Zigbee message from 'WZDeskLamp', type 'attributeReport', cluster 'genOnOff', data '{"onOff":0}' from endpoint 1 with groupID null
MQTT publish: topic 'zigbee2mqtt/WZDeskLamp', payload '{"linkquality":255,"state":"OFF"}'
Received Zigbee message from 'WZDeskLamp', type 'attributeReport', cluster 'genOnOff', data '{"onOff":1}' from endpoint 1 with groupID null
MQTT publish: topic 'zigbee2mqtt/WZDeskLamp', payload '{"linkquality":255,"state":"ON"}'
```
Whats not working is to set the state (turn it on/off) via mqtt:
```
Received MQTT message on 'zigbee2mqtt/AZDesktop/1/set' with data '{"state":"ON"}'
No converter available for 'state' ("ON")
Received MQTT message on 'zigbee2mqtt/AZDesktop/1/set' with data '{"state":"OFF"}'
No converter available for 'state' ("OFF")
```

Did add following:
```
external_converters:
  - 404117.js
```
to my configuration.yaml

My external converter (404117.js):
```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['Desk lamp'], // The model ID from: Device with modelID 'lumi.sens' is not supported.
    model: '404117', // Vendor model number, look on the device for a model number
    vendor: 'Müller Licht', // Vendor of the device (only used for documentation and startup logging)
    description: 'Müller Licht Nolia desk lamp', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
    fromZigbee: [fz.on_off, fz.command_on, fz.command_off, fz.command_toggle], // We will add this later
    toZigbee: [tz.on_off], // Should be empty, unless device can be controlled (e.g. lights, switches).
    exposes: [e.action(['ON', 'OFF'])], // Defines what this device exposes, used for e.g. Home Assistant discovery and in the frontend
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'genBasic']);
        await reporting.onOff(endpoint);
    }
};

module.exports = definition;
```

Any idea why it does not find the converter? It keeps telling "No converter available for 'state' ("OFF")"

Thanks and best regards
Dom

**Comment by lastfreeusernameicouldfind** [comment]
same for zigbee2mqtt/AZDesktop/set
```
Received MQTT message on 'zigbee2mqtt/AZDesktop/set' with data '{"state":"ON"}'
Error 2023-12-06 21:19:28No converter available for 'state' ("ON")
Debug 2023-12-06 21:19:29Received MQTT message on 'zigbee2mqtt/AZDesktop/set' with data '{"state":"OFF"}'
Error 2023-12-06 21:19:29No converter available for 'state' ("OFF")
```

**Comment by lastfreeusernameicouldfind** [comment]
Should the above work or am I doing something wrong?

**Reply by rakete4711** [reply]
Hi Dom,

 here is my working converter:

`const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const legacy = require('zigbee-herdsman-converters/lib/legacy');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['Desk lamp'],
    model: '404117',
    vendor: 'Müller Licht',
    description: 'Müller Licht Nolia desk lamp',
    // Note that fromZigbee, toZigbee and exposes are missing here since we use extend here.
    // Extend contains a default set of fromZigbee/toZigbee converters and expose for common device types.
    // The following extends are available:
    // - extend.switch
    // - extend.light_onoff_brightness
    // - extend.light_onoff_brightness_colortemp
    // - extend.light_onoff_brightness_color
    // - extend.light_onoff_brightness_colortemp_color
    extend: extend.light_onoff_brightness_colortemp_color(),
};

module.exports = definition;

`

**Reply by lastfreeusernameicouldfind** [reply]
Hi rakete4711,

thanks.
on/off works by sending `zigbee2mqtt/WZDeskLamp/set' with data '{"state":"ON"}`

Which command does work for you to adjust color_temp or color?
When I am setting brightness I am always getting:
`Received MQTT message on 'zigbee2mqtt/WZDeskLamp/set' with data '{"state":"on","brightness":"95"}'
Error: Cannot read properties of undefined (reading 'includes')`

For: color_temp:
`Received MQTT message on 'zigbee2mqtt/WZDeskLamp/set' with data '{"color_temp": 154 , "color_temp": 500}'
Error: Cannot read properties of undefined (reading 'includes')`

For color:
`Received MQTT message on 'zigbee2mqtt/WZDeskLamp/set' with data '{"color":{"b":71,"g":255,"r":71},"transition":1}'
Error: Cannot read properties of undefined (reading 'includes')`


Thanks allot for your help,
Dom

**Reply by rakete4711** [reply]
Hi Dom,

I'm using for further actions the light card in the home assistant dashboard. When I set the brightness to another level, I get the follow messages in the Z2M Log:
Debug 2023-12-31 14:17:14Publishing 'set' 'brightness' to 'Tischleuchte Nolia'
Info 2023-12-31 14:17:14MQTT publish: topic 'zigbee2mqtt/Tischleuchte Nolia', payload '{"brightness":120,"color_mode":"color_temp","color_temp":370,"linkquality":15,"power_on_behavior":null,"state":"ON"}'

**Reply by rakete4711** [reply]
This works in my case:
Received MQTT message on 'zigbee2mqtt/Tischleuchte Nolia/set' with data '{"state":"ON","brightness":170}'

**Reply by rakete4711** [reply]
look at the difference:
your command: /set' with data '{"state":"on","brightness":"95"}'
my command:   /set' with data '{"state":"ON","brightness":170}'
you wrote the value for brightness within "" and ON in low letters

**Reply by lastfreeusernameicouldfind** [reply]
oh man thanks! There was another issue in my converter. Now its working. Thanks allot!
